presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1192,27 +1194,30 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
             int bucketCount,
             PartitionUpdate partitionUpdate)
     {
-        if (partitionUpdate.getFileNames().size() == bucketCount) {
+        if (partitionUpdate.getStagingFileInfos().size() == bucketCount) {
             // fast path for common case
             return ImmutableList.of();
         }
         HdfsContext hdfsContext = new HdfsContext(session, table.getDatabaseName(), table.getTableName());
         JobConf conf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, targetPath));
         String fileExtension = HiveWriterFactory.getFileExtension(conf, fromHiveStorageFormat(storageFormat));
-        Set<String> fileNames = ImmutableSet.copyOf(partitionUpdate.getFileNames());
-        ImmutableList.Builder<String> missingFileNamesBuilder = ImmutableList.builder();
+        Set<String> targetFileNames = ImmutableSet.copyOf(
+                partitionUpdate.getStagingFileInfos().stream()
+                        .map(StagingFileInfo::getTargetFileName)
+                        .collect(toImmutableSet()));
+        ImmutableList.Builder<StagingFileInfo> missingFileNamesBuilder = ImmutableList.builder();
         for (int i = 0; i < bucketCount; i++) {
-            String fileName = HiveWriterFactory.computeBucketedFileName(filePrefix, i) + fileExtension;
-            if (!fileNames.contains(fileName)) {
-                missingFileNamesBuilder.add(fileName);
+            String targetFileName = HiveWriterFactory.computeBucketedFileName(filePrefix, i) + fileExtension;
+            if (!targetFileNames.contains(targetFileName)) {
+                missingFileNamesBuilder.add(new StagingFileInfo(targetFileName, targetFileName));
```

Comment:
What's the purpose of this (instead of returning a `List<String>` ? :)  -- Looks like they get called into `createEmptyFile` anyway, which can take `List<String>`

Reply:
The problem here is we need to merge empty partition's `PartitionUpdate` with non-empty partitions's `PartitionUpdate` for later processing. See `computePartitionUpdatesForMissingBuckets`.